internal/resource/cuda-device.go

```diff
@@ -96,3 +96,7 @@ func (d *cudaDevice) IsMigCapable() (bool, error) {
 func (d *cudaDevice) IsMigEnabled() (bool, error) {
 	return false, nil
 }
+
+func (d *cudaDevice) GetClass() (string, error) {
```

Comment:
Question: Does it make sense for the class to be an `int` instead? Do we really need to make a string comparison?

Reply:
Used `uint32` because thats what nv-golib is returning